extensions/cornerstone/src/getSopClassHandlerModule.js

```diff
@@ -102,13 +108,43 @@ function _getDisplaySetsFromSeries(instances, servicesManager, extensionManager)
     numInstances: 1,
     imageIdForThumbnail, // thumbnail image
     others: instances, // all other level instances in the image Pyramid
+    instances,
     othersFrameOfReferenceUID,
+    imageIds: instances.map(instance => instance.imageId),
   };
+  // The microscopy viewer directly accesses the metadata already loaded, and
+  // uses the DICOMweb client library directly for loading, so it has to be
+  // provided here.
+  const dicomWebClient = dataSource.retrieve.getWadoDicomWebClient?.();
+  const instanceMap = new Map();
+  instances.forEach(instance => instanceMap.set(instance.imageId, instance));
+  if (dicomWebClient) {
+    // This is really ugly - maybe this should be stored unmodified.
+    dicomWebClient.getDICOMwebMetadata = imageId => transferDenaturalizedDataset(denaturalizeDataset(fixMultivalueKeys(instanceMap.get(imageId))));
+
+    csUtils.genericMetadataProvider.add(displaySet.imageIds[0], {
+      type: 'webClient',
+      rawMetadata: dicomWebClient,
+    });
+  } else {
+    // TODO - handle other viewer types by simulating the viewer
+    console.warn("Unable to provide a DICOMWeb client library, microscopy will fail to view");
+  }
 
   return [displaySet];
 }
 
-export default function getDicomMicroscopySopClassHandler({ servicesManager, extensionManager }) {
+/** Fix multi-valued keys before denaturalizing */
+export function fixMultivalueKeys(naturalData, keys = ['ImageType']) {
+  for (const key of keys) {
+    if (typeof naturalData[key] === 'string') {
+      naturalData[key] = naturalData[key].split('\\');
+    }
+  }
+  return naturalData;
+}
```

Comment:
Does it make sense to put it in the dicomUtils? `import { dicomWebUtils } from '@ohif/extension-default';`